src/newcharacter.cpp

```diff
@@ -535,32 +535,13 @@ bool player::create(character_type type, std::string tempname)
         }
     }
 
-    std::vector<item> prof_items = g->u.prof->items( g->u.male );
-
-    // Those with certain special traits are guaranteed to start with certain items
-    if( has_trait( "HYPEROPIC" ) && has_trait( "MYOPIC" ) ) {
-        prof_items.push_back( item( "glasses_bifocal" ) );
-    }
-    else if( has_trait( "MYOPIC" ) ) {
-        prof_items.push_back( item( "glasses_eye" ) );
-    }
-    else if( has_trait( "HYPEROPIC" ) ) {
-        prof_items.push_back( item( "glasses_reading" ) );
-    }
-
-    if( has_trait( "ASTHMA" ) ) {
-        prof_items.push_back( item( "inhaler", 0, item::default_charges_tag{} ) );
-    }
-    if (has_trait("CANNIBAL")) {
-        prof_items.push_back( item( "cookbook_human", 0 ) );
-    }
-    if( has_trait( "ALBINO" ) ) {
-        prof_items.push_back( item( "teleumbrella", 0 ) );
-    }
+    std::list<item> prof_items = g->u.prof->items( g->u.male, g->u.get_mutations() );
 
     for( item &it : prof_items ) {
         if( it.is_armor() ) {
-            wear_item( it, false ); // If wearing fails, we fail silently
```

Comment:
I see, the PR's title does imply that. Also, since `json_item_substitution::load` works by appending entries, mods will be able to add additional substitutions as needed.